### PR TITLE
Add docs demonstrating how to share a Workspace with Sidecars

### PR DESCRIPTION
**Pull request by sbwsg**
# Changes

Prior to this commit we didn't have any documentation showing how to share
a Workspace between Steps of a Task and the Task's Sidecars. In fact
there were two bugs that prevented this from working correctly. Those
bugs have since been squashed.

This commit adds documentation and an example yaml showing how to share a
Workspace between a Task's Steps and its Sidecars.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
Added an example and documentation showing how to share a Workspace between Steps and Sidecars in a Task.
```

/kind documentation

### CODE REVIEW

**Comment by sbwsg** [comment]
This commit depends on two bugs being fixed: https://github.com/tektoncd/pipeline/pull/3315 and https://github.com/tektoncd/pipeline/pull/3318

**Comment by sbwsg** [comment]
/hold

... until those bugs are fixed and merged.

**Comment by sbwsg** [comment]
Integration failures are accurate but fixed by PRs mentioned in above comment.

**Comment by sbwsg** [comment]
/hold cancel 

both dependent bugs are fixed now.

**Comment by sbwsg** [comment]
It looks like the TaskRun's pod is timing out before the first step gets to run its few lines of script.

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**
